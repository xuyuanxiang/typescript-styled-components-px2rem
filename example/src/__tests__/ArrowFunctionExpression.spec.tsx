import React from 'react';
import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import {
  ArrowFunctionExpressionWithPureBody,
  ArrowFunctionWithBinaryBody,
  ArrowFunctionWithBlockBody,
  ArrowFunctionWithConditionalBody,
} from '../ArrowFunctionExpression';

describe('embedded ArrowFunctionExpression', () => {
  const div: HTMLDivElement = document.createElement('div');

  beforeEach(() => {
    document.body.appendChild(div);
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
    document.body.removeChild(div);
  });

  it('should transform PropertyAccess, Identifier, StringLiteral, NumericLiteral Body', function() {
    TestUtils.act(() => {
      ReactDOM.render(<ArrowFunctionExpressionWithPureBody id="inputId" width={320} />, div);
    });
    const input = div.querySelector('input#inputId');
    if (input) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(input);
      expect(style.color).toBe('palevioletred');
      expect(style.border).toBe('0.01rem solid palevioletred');
      expect(style.borderRadius).toBe('0.08rem');
      expect(style.width).toBe('3.2rem');
      expect(style.padding).toBe('0px');
      expect(style.margin).toBe('0.32rem');
      expect(style.lineHeight).toBe('0.44rem');
      expect(style.height).toBe('0.44rem');
    } else {
      throw new Error('ArrowFunctionExpressionWithPureBody should be render');
    }
  });

  it('should transform BinaryExpression Body', function() {
    TestUtils.act(() => {
      ReactDOM.render(<ArrowFunctionWithBinaryBody id="buttonId" marginVertical={24} disabled />, div);
    });
    const button = div.querySelector('button#buttonId');
    if (button) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(button);
      expect(style.margin).toBe('0.24rem 0.08rem 0.24rem 0rem');
      expect(style.border).toBe('0.02rem solid darkgray');
      expect(style.borderRadius).toBe('0.2rem');
      expect(style.fontSize).toBe('0.18rem');
      expect(style.color).toBe('black');
    } else {
      throw new Error('ArrowFunctionWithBinaryBody should be render');
    }
  });

  it('should transform ConditionalExpression Body', function() {
    TestUtils.act(() => {
      ReactDOM.render(<ArrowFunctionWithConditionalBody id="buttonId" width={280} />, div);
    });
    const button = div.querySelector('button#buttonId');
    if (button) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(button);
      expect(style.width).toBe('2.8rem');
      expect(style.height).toBe('2rem');
      expect(style.fontSize).toBe('0.18rem');
      expect(style.color).toBe('black');
    } else {
      throw new Error('ArrowFunctionWithConditionalBody should be render');
    }
  });

  it('should transform Block Body', function() {
    TestUtils.act(() => {
      ReactDOM.render(<ArrowFunctionWithBlockBody id="buttonId" width={320} />, div);
    });
    const button = div.querySelector('button#buttonId');
    if (button) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(button);
      expect(style.width).toBe('3.2rem');
      expect(style.fontSize).toBe('0.18rem');
      expect(style.color).toBe('black');
    } else {
      throw new Error('ArrowFunctionWithBlockBody should be render');
    }
  });
});
