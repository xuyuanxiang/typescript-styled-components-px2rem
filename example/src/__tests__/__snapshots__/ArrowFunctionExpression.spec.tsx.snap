// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embedded ArrowFunctionExpression should transform BinaryExpression Body 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.hRVDXl{color:palevioletred;border:0.01rem solid palevioletred;border-radius:0.08rem;width:3.2rem;height:0.44rem;line-height:0.44rem;margin:0.32rem;padding:0;}
      
/* sc-component-id: sc-bwzfXH */

      
/* sc-component-id: sc-htpNat */
.hKHuWL{font-size:0.18rem;color:black;margin:0.24rem 0.08rem 0.24rem 0rem;border:0.02rem solid darkgray;border-radius:0.2rem;}
      
/* sc-component-id: sc-bxivhb */

    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-htpNat hKHuWL"
        disabled=""
        id="buttonId"
      />
    </div>
  </body>
</html>
`;

exports[`embedded ArrowFunctionExpression should transform Block Body 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.hRVDXl{color:palevioletred;border:0.01rem solid palevioletred;border-radius:0.08rem;width:3.2rem;height:0.44rem;line-height:0.44rem;margin:0.32rem;padding:0;}
      
/* sc-component-id: sc-bwzfXH */
.enWAqc{font-size:0.18rem;color:black;width:3.2rem;}
      
/* sc-component-id: sc-htpNat */
.hKHuWL{font-size:0.18rem;color:black;margin:0.24rem 0.08rem 0.24rem 0rem;border:0.02rem solid darkgray;border-radius:0.2rem;}
      
/* sc-component-id: sc-bxivhb */
.cYEBPM{font-size:0.18rem;color:black;width:2.8rem;height:2rem;}
    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-bwzfXH enWAqc"
        id="buttonId"
        width="320"
      />
    </div>
  </body>
</html>
`;

exports[`embedded ArrowFunctionExpression should transform ConditionalExpression Body 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.hRVDXl{color:palevioletred;border:0.01rem solid palevioletred;border-radius:0.08rem;width:3.2rem;height:0.44rem;line-height:0.44rem;margin:0.32rem;padding:0;}
      
/* sc-component-id: sc-bwzfXH */

      
/* sc-component-id: sc-htpNat */
.hKHuWL{font-size:0.18rem;color:black;margin:0.24rem 0.08rem 0.24rem 0rem;border:0.02rem solid darkgray;border-radius:0.2rem;}
      
/* sc-component-id: sc-bxivhb */
.cYEBPM{font-size:0.18rem;color:black;width:2.8rem;height:2rem;}
    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-bxivhb cYEBPM"
        id="buttonId"
        width="280"
      />
    </div>
  </body>
</html>
`;

exports[`embedded ArrowFunctionExpression should transform PropertyAccess, Identifier, StringLiteral, NumericLiteral Body 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.hRVDXl{color:palevioletred;border:0.01rem solid palevioletred;border-radius:0.08rem;width:3.2rem;height:0.44rem;line-height:0.44rem;margin:0.32rem;padding:0;}
      
/* sc-component-id: sc-bwzfXH */

      
/* sc-component-id: sc-htpNat */

      
/* sc-component-id: sc-bxivhb */

    </style>
  </head>
  <body>
    <div>
      <input
        class="sc-bdVaJa hRVDXl"
        id="inputId"
        type="password"
        width="320"
      />
    </div>
  </body>
</html>
`;
